### PR TITLE
Failing tests for coercion of numeric strings to numbers

### PR DESCRIPTION
**Pull request by taras**
## Hey, I just made a Pull Request!

Added failing tests for numeric number coercion as per https://discordapp.com/channels/687207715902193673/687235481154617364/750922915721117709

#### :heavy_check_mark: Checklist

- [ ] All tests are passing `yarn test`
- [ ] Screenshots attached (for UI changes)
- [ ] Relevant documentation updated
- [ ] Prettier run on changed files
- [x] Tests added for new functionality
- [ ] Regression tests added for bug fixes

### CODE REVIEW

File: packages/config/src/reader.test.ts [hunk 1]
```diff
@@ -35,6 +35,7 @@ const DATA = {
     strings: ['string1', 'string2'],
   },
   nestlings: [{ boolean: true }, { string: 'string' }, { number: 42 }] as {}[],
+  port: 'number',
```
Comment (Rugvip) [inline packages/config/src/reader.test.ts:38]: This bit seems a bit strange?

Comment (taras) [reply packages/config/src/reader.test.ts:38]: Yeah, I'm pre-coffee. I thought this was schema. I'll change it to something else.

Comment (Rugvip) [reply packages/config/src/reader.test.ts:38]: Coding in the Danger Zone 🎶

Comment (taras) [reply packages/config/src/reader.test.ts]: I removed that line. It's not really doing anything.

**Review by Rugvip — APPROVED** [review]
This is a great workaround for sneaking stuff into master :grin:

**Comment by Rugvip** [comment]
:shipit:

**Pull request merged**